lib/MetaCPAN/Web/Controller/Root.pm

```diff
@@ -103,6 +106,15 @@ sub end : ActionClass('RenderView') {
 
     $c->stash->{site_alert_message} = $c->config->{site_alert_message};
 
+    my $file = $c->config->{home} . '/events.json';
+    eval {
+        my $all_ads = decode_json( path($file)->slurp_utf8 );
```

Comment:
This should be `slurp` (or `slurp_raw`).  `decode_json` expects UTF-8 bytes, not a character string.